### PR TITLE
[wip] Use proc.communicate to avoid hanging

### PR DESCRIPTION
**Pull request by richardliaw**
This replaces a loop of reading from PIPE and dealing with odd buffer underflow issues
by using proc.communicate. 

These issues seem to only come up for memory tests, where the output of the
command actually does not matter at all.

### CODE REVIEW

**Comment by pcmoritz** [comment]
Thanks for doing this, I'll try it out/review later tonight after I have a patch for the pytorch issue.

File: asv/util.py [hunk 1]
```diff
@@ -513,42 +513,22 @@ def sig_forward(signum, frame):
...
-                        list(fds.keys()), [], [], timeout)
-                except select.error as err:
-                    if err.args[0] == errno.EINTR:
-                        # interrupted by signal handler; try again
-                        continue
-                    raise
-
-                if len(rlist) == 0:
-                    # We got a timeout
+                    result = proc.communicate(timeout=10)
```
Comment (pschafhalter) [inline asv/util.py:517]: Use the timeout argument?

**Review by pschafhalter — CHANGES_REQUESTED** [review]
Looks like this works and there's no hanging on the actor benchmarks!

Maybe set timeout=10 in benchmark.py as well? Otherwise timed out tests fail with "asv: benchmark timed out (timeout 60.0s)" despite timeout being 10.0s.

**Comment by pcmoritz** [comment]
For me, both benchmark_task.CPUTaskSuite.peakmem_task_lifecycle and benchmark_task.CPUTaskSuite.time_task_dependency are timing out with these setting, can we fix that before merging?

**Comment by pcmoritz** [comment]
My suggestion for fixing the problem is here: https://github.com/ray-project/asv/pull/4

**Comment by pschafhalter** [comment]
Fixed with #4

**Pull request closed without merging**